Add ability to defer fetching a value in the fold - but without requiring Journal check

martinsumner opened the pull request:
…ing check presence

Reduces the cost of using head folds in partition repairs, where the check is not necessary as a fetch failure will be caught and prompt read repair.

martinsumner committed:
Add ability to defer fetching a value in the fold - but without requring check presence

Reduces the cost of using head folds in partition repairs, where the check is not necessary as a fetch failure will be caught and prompt read repair.

stephenwood24-nhs commented on src/leveled_runner.erl:
Is the spec for `accumulate_objects` correct as DeferredFetch could be `{true, {defer, _}}`
Also is this passed around other functions, if so do their specs need updating?

martinsumner replied:
I will update the specs in the leveled_runner module.

martinsumner committed:
Update types for JournalCheck

martinsumner committed:
Fix tuple mistake (#473)

Otherwise leveled_pclerk will crash if there is a partial merge that results in an an empty addition
